lib/chef/resource/windows_pagefile.rb

```diff
@@ -102,20 +113,35 @@ class WindowsPagefile < Chef::Resource
       end
 
       action :delete, description: "Deletes the specified pagefile." do
-        validate_name
-        delete(new_resource.path) if exists?(new_resource.path)
+        pagefile = clarify_pagefile_name
+        delete(pagefile ) if exists?(pagefile )
       end
 
       action_class do
         private
 
-        # make sure the provided name property matches the appropriate format
-        # we do this here and not in the property itself because if automatic_managed
-        # is set then this validation is not necessary / doesn't make sense at all
+        def clarify_pagefile_name
+          if new_resource.path.length < 4
+            (new_resource.path[0] + ":\\pagefile.sys")
+          elsif new_resource.path !~ /^.:.*/
+            "C:\\pagefile.sys"
```

Comment:
that seems a bit weird, people could have gross typos and it'd just give them a pagefile on c

feels a bit more like you should have validation that the path takes the form of:

1.  bare drive letter
2.  bare drive letter with colon
3.  bare drive letter with colon and one (or more?) backwhacks
4.  bare drive letter with colon and backwhack and arbitrarily long path

and if its not one of those it should raise rather than having a default

the bare drive letter should also be validated into being [A-Za-z] so that `1:` wasn't valid and we didn't try to create a pagefile on the "1" drive (or unicode pasted in from some blog post somewhere).

Reply:
oh yeah you've got validate_name, but i'd do that sooner and be a bit more strict.  you're also validating post-process and munging into the full path, and giving the user an error message about that, so the user that has passes in an invalid drive letter gets back an error message right now which has extraneous information about the path.

this could also probably be a `coerce` on the resource instead of being in the action.

Reply:
updated!